### PR TITLE
[String] Fix of title() method

### LINKED ISSUE #34045: [String] Bug in title() method with Spanish
String component doesn't process Spanish correctly
```(new UnicodeString('¡hola!'))->title()```
returns `¡hola!` instead of expected `¡Hola!`

### PR DESCRIPTION
**Pull request by alex-bacart**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #34045
| License       | MIT

Fix of `title()` method for Spanish

### CODE REVIEW

File: src/Symfony/Component/String/ByteString.php [hunk 1]
```diff
@@ -368,7 +368,12 @@ public function startsWith($prefix): bool
     public function title(bool $allWords = false): parent
     {
         $str = clone $this;
-        $str->string = $allWords ? ucwords($str->string) : ucfirst($str->string);
+
+        $limit = $allWords ? -1 : 1;
+
+        $str->string = preg_replace_callback('/\b./u', static function ($m): string {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/String/ByteString.php]: The `u` modifier must be removed `'/\b./'` - which means your test case should be moved to `AbstractUnicodeTestCase`. I'm not sure we can implement this change on `ByteString`: a high-ASCII byte might be a letter actually. It all depends on how `\b` figures out the word boundary.

File: src/Symfony/Component/String/Tests/AbstractAsciiTestCase.php [hunk 2]
```diff
@@ -495,6 +495,8 @@ public static function provideTitle()
             ['Symfony', 'symfony', false],
             ['Symfony', 'Symfony', false],
             ['SYmFOny', 'sYmFOny', false],
+            ['¡Hola spain!', '¡hola spain!', false],
+            ['¡Hola Spain!', '¡hola spain!', true],
```
Comment (nicolas-grekas) [inline src/Symfony/Component/String/Tests/AbstractAsciiTestCase.php]: Can you also add a test case where the first letter is one with an accent?

File: src/Symfony/Component/String/ByteString.php (hunk 1 above)
Comment (alex-bacart) [reply src/Symfony/Component/String/ByteString.php]: You are right, removed these code from `ByteString`. And moved test cases to `AbstractUnicodeTestCase`.

File: src/Symfony/Component/String/Tests/AbstractAsciiTestCase.php (hunk 2 above)
Comment (alex-bacart) [reply src/Symfony/Component/String/Tests/AbstractAsciiTestCase.php]: Done!

Comment (nicolas-grekas) [reply src/Symfony/Component/String/Tests/AbstractAsciiTestCase.php]: I can't see this, where?
Here is an example: `être ou ne pas être`

Comment (alex-bacart) [reply src/Symfony/Component/String/Tests/AbstractAsciiTestCase.php]: Sorry I didn't understand you for the first time. Added.

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @alex-bacart.

**Pull request merged**